src/middlewares/confirmationMiddleware.js

```diff
@@ -0,0 +1,40 @@
+import response from '../utils/response';
+import messages from '../utils/messages';
+import models from '../models';
+
+const { Request, User } = models;
+
+/**
+ * ReConfirm Travel Request Middleware
+ * @param {Object} req - Express request object
+ * @param {Oject} res - Express response object
+ * @param {Function} next - Express next middleware function
+ * @returns {Object} - Returns Object
+ */
+const requestConfirmation = async (req, res, next) => {
+  const { requestId } = req.params;
+  let { confirmation } = req.query;
+  confirmation = (confirmation === 'true');
+  try {
+    if (!confirmation) {
+      const option = {
+        include: [{
+          model: User,
+          as: 'User',
+          attributes: ['firstName', 'lastName']
+        }],
+        attributes: { exclude: ['id', 'approvalStatus', 'userId', 'createdAt', 'updatedAt', 'multiCity'] }
+      };
+      const request = await Request.findByPk(requestId, option);
```

Comment:
is error handled if request is not found?

Reply:
Yes, it's handled. There are middlewares in the endpoints before this, that checks requestId.

Reply:
Okay